src/component.jl

```diff
@@ -208,14 +214,14 @@ function get_time_series_values(
     len::Union{Nothing, Int} = nothing,
     count::Int = 1,
 ) where {T <: TimeSeriesData}
-    return TimeSeries.values(get_time_series_array(
+    return Array{Float64}(TimeSeries.values(get_time_series_array(
```

Comment:
I think there are some more fundamental problems here.  The `get_time_series_*` functions are designed to work with `SingleTimeSeries`.  I'm surprised anything works with `Deterministic`.  We shouldn't be calling `TimeSeries` methods on the data object for `Deterministic` because it is a SortedDict of DateTime to Array.  Also, it will fail if `scaling_factor_multiplier` is set.

This change should be reverted for `SingleTimeSeries` because the data will already be an array.  We need a new layer of internal methods to handle differences between the types.

Reply:
Discussion from today:  The `get_time_series_*` functions will not accept count.  For forecasts they will always return one window.